lib/MusicBrainz/Server/Edit/Medium/Edit.pm

```diff
@@ -301,15 +301,6 @@ sub build_display_data
             $data->{changed_mbids} = 1;
         }
 
-        $data->{artist_credit_changes} = [
```

Comment:
This and `hash_artist_credit_without_join_phrases` weren't used outside of tests, so I moved what's here into the test file, but without the join phrase exclusion logic, because that isn't required by what the tests are checking (they basically just check the length of `artist_credit_changes` and the type of changes).